lib/hammer_cli_foreman/command_extensions/ping.rb

```diff
@@ -10,9 +10,28 @@ class Ping < HammerCLI::CommandExtensions
         end
       end
 
+      def self.check_for_unrecognized(plugins, output_definition)
+        failed = plugins.select { |_, data| data['services'] }
+                        .each_with_object([]) { |(_, d), s| s << d['services'] }
+                        .reduce({}, :merge)
+                        .select do |name, data|
+          begin
+            output_definition.find_field(name)
+            false
+          rescue ArgumentError
+            data['status'] == _('FAIL')
+          end
+        end
```

Comment:
This logic is quite hard to follow. Why does it modify data in place? Shouldn't it be a more pure function that returns a list of unrecognized services?

Reply:
This method doesn't modify anything. If there is a failed service that has no field (won't be shown) it simply puts a warning. Otherwise does nothing additional.